libcaliptra/src/caliptra_api.c

```diff
@@ -364,13 +371,19 @@ bool caliptra_ready_for_firmware(void)
 uint32_t caliptra_ready_for_runtime(void)
 {
     uint32_t status;
+    uint32_t fatal_error;
     bool ready = false;
 
     do
     {
         status = caliptra_read_status();
+        fatal_error = caliptra_read_fw_fatal_error();
 
-        if ((status & GENERIC_AND_FUSE_REG_CPTRA_FLOW_STATUS_READY_FOR_FW_MASK) == GENERIC_AND_FUSE_REG_CPTRA_FLOW_STATUS_READY_FOR_FW_MASK)
+        if (fatal_error != 0)
+        {
+            return fatal_error;
+        }
+        else if ((status & GENERIC_AND_FUSE_REG_CPTRA_FLOW_STATUS_READY_FOR_FW_MASK) == GENERIC_AND_FUSE_REG_CPTRA_FLOW_STATUS_READY_FOR_FW_MASK)
```

Comment:
I think there might have been a bug here previously. Isn't this supposed to be `GENERIC_AND_FUSE_REG_CPTRA_FLOW_STATUS_READY_FOR_RUNTIME_MASK`?

Reply:
Done